docs/internals/layout_in_memory.rst

```diff
@@ -5,36 +5,121 @@
 Layout in Memory
 ****************
 
+Reserved Memory Areas
+=====================
+
 Solidity reserves four 32-byte slots, with specific byte ranges (inclusive of endpoints) being used as follows:
 
 - ``0x00`` - ``0x3f`` (64 bytes): scratch space for hashing methods
-- ``0x40`` - ``0x5f`` (32 bytes): currently allocated memory size (aka. free memory pointer)
+- ``0x40`` - ``0x5f`` (32 bytes): allocated memory size (aka. free memory pointer)
 - ``0x60`` - ``0x7f`` (32 bytes): zero slot
 
-Scratch space can be used between statements (i.e. within inline assembly). The zero slot
-is used as initial value for dynamic memory arrays and should never be written to
-(the free memory pointer points to ``0x80`` initially).
+Memory Allocation
+-----------------
+
+Solidity compiles to EVM instructions, and implements its complex data types on top of
+EVM features. Memory allocation is not an EVM feature. Inestead, Solidity implements its
+memory allocation via the free memory pointer stored at location ``0x40``. That value tracks the next
```

Comment:
Could be misleading - you could read that in a way that it only tracks a single 32-byte memory "slot".